drivers/hwinfo/hwinfo_esp32.c

```diff
@@ -13,7 +13,7 @@
 #include <zephyr/drivers/hwinfo.h>
 #include <string.h>
 
-ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
+int z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
```

Comment:
This function could fail if a parameter were invalid.

But then again, it should probably also fail if the device ID exceeds `INT32_MAX` 😅.

It's probably safe.

Reply:
I hate `size_t`, it is pointless type. `sizeof` should be returning `unsigned int`.

Reply:
The C standard only states that `size_t` should have conversion rank not greater than `signed long` (+ the implementation specific magic words), so it does not seem that there is easy way to find out what is best arithmetic type equivalent. So `int` it is.

Reply:
`size_t` is standard C (not POSIX), and is used pretty ubiquitously this way. It's not really a concern in this PR which is just addressing the layering violation. If `int` is otherwise fine, should this comment be marked as resolved?

Reply:
> `size_t` is standard C (not POSIX), and is used pretty ubiquitously this way. It's not really a concern in this PR which is just addressing the layering violation. If `int` is otherwise fine, should this comment be marked as resolved?

Yes, I know. It just shows how the language standard is out of touch with reality.